src/MAX30105.cpp

```diff
@@ -361,29 +361,55 @@ uint8_t MAX30105::getReadPointer(void) {
 }
 
 
+void MAX30105::startTempMeasurement()
+{
+	// setting appropriate bit to start temp measurement
+	writeRegister8(_i2caddr, MAX30105_DIETEMPCONFIG, 0x01);
+}
+
+bool MAX30105::getTemperature(float& temperature)
+{
+	uint8_t isTempMeasurementReady = readRegister8(_i2caddr, MAX30105_INTSTAT2);
+	if (isTempMeasurementReady & MAX30105_INT_DIE_TEMP_RDY_ENABLE)
+	{
+		// temp measurement available!
+		temperature = readTemperature(true);
+		return true;
+	}
+	else
+	{
+		// busy converting
+		return false;
+	}
+}
+
 // Die Temperature
 // Returns temp in C
-float MAX30105::readTemperature() {
+float MAX30105::readTemperature(bool isAsync) {
```

Comment:
I am going to keep the functions as is since it has been tested working and any major change will require another round of testing. But I will consider this for future changes.